packages/blade/src/components/Typography/BaseText/BaseText.web.tsx

```diff
@@ -14,6 +14,7 @@ const StyledBaseText = styled.div<StyledBaseTextProps>(
     fontWeight,
     fontStyle,
     textDecorationLine,
+    numberOfLines,
```

Comment:
Moved the truncation logic from Text component to BaseText because: 

Previously it was like this: 

```jsx
// Text.tsx file
const StyledText = styled(BaseText)`
  // truncation logic
`

<StyledText as="p" />
```

Now the problem is that if you directly pass as="p" to styledText it will just replace the BaseText with "p" component which will have default browser styling and doesn't use the BaseText anymore, to circumvent this I had to mess around with all sorts of StyledComponent bullshit and forward the as via `[forwardedAs](https://styled-components.com/docs/api#forwardedas-prop)` prop which was getting complex. 

It's just much simpler to move the logic to BaseText instead.